collect_app/src/main/java/org/odk/collect/android/widgets/items/SelectOneFromMapDialogFragment.kt

```diff
@@ -87,20 +93,47 @@ class SelectOneFromMapDialogFragment : MaterialFullScreenDialogFragment(), Fragm
 
     companion object {
         const val ARG_FORM_INDEX = "form_index"
+        const val ARG_SELECTED_INDEX = "selected_index"
     }
 }
 
-internal class SelectChoicesMapData(private val resources: Resources, prompt: FormEntryPrompt) :
+internal class SelectChoicesMapData(
+    private val resources: Resources,
+    scheduler: Scheduler,
+    prompt: FormEntryPrompt,
+    private val selectedIndex: Int?
```

Comment:
We need to pass the selected index through because the widget answer will be "dirty" (in the widget but not in the `FormEntryPrompt`) when return from the map. We ended up with something similar in the select minimal implementation - we pass through the current answer rather than relying on the `FormEntryPrompt`.